WebService/Auth/AuthMiddleware.cs

```diff
@@ -181,6 +191,20 @@ private bool ValidateToken(string token, HttpContext context)
                     // header doesn't need to be parse again later in the User controller.
                     context.Request.SetCurrentUserClaims(jwtToken.Claims);
 
+                    // Store the user allowed actions in the request context to validate
+                    // authorization later in the controller.
+                    var userObjectId = context.Request.GetCurrentUserObjectId();
+                    var roles = context.Request.GetCurrentUserRoleClaim().ToList();
+                    if (roles.Any())
+                    {
+                        var allowedActions = this.userManagementClient.GetAllowedActionsAsync(userObjectId, roles).Result;
+                        context.Request.SetCurrentUserAllowedActions(allowedActions);
+                    }
+                    else
+                    {
+                        this.log.Error("JWT token doesn't include any role claims.", () => { });
```

Comment:
Is this really an error or do we default to read only in this case?

Reply:
We do default to read-only. In iot hub manager we log an error, I think warn might make more sense. @ppathan @zhang-hua thoughts?

Reply:
I think warn makes more sense if we are returning true.

Reply:
@molly-moen @ppathan : this was intentionally for the user which don't have any assigned roles by default.  make more sense to use warn().